src/common/tests_dmg_helpers.c

```diff
@@ -615,7 +610,9 @@ parse_device_info(struct json_object *smd_dev, device_list *devices,
 			D_ERROR("unable to extract state from JSON\n");
 			return -DER_INVAL;
 		}
-		strcpy(devices[*disks].state, json_object_to_json_string(tmp));
+
+		strncpy(devices[*disks].state, json_object_to_json_string(tmp),
+			sizeof(devices[*disks].state) - 1);
```

Comment:
Is json_object_to_json_string null terminated? Same question as above why subtracting 1 from sizeof?

Reply:
I think so. I tested it locally and it worked.